src/onelogin/aws-assume-role/aws-assume-role.py

```diff
@@ -442,12 +446,16 @@ def main():
                     print("-----------------------------------------------------------------------")
                     print("Select the desired AWS Role [0-%s]: " % (len(roles) - 1))
 
-                    role_option = get_selection(len(roles))
+                    if options.aws_role_index:
+                        role_option = options.aws_role_index
```

Comment:
I'm not sure that what you need to pass an option is an index, instead of the name of the role.

So here it should search if the role name exists as an option and then take its index, otherwise notify that the previously selected role is not available so ask for select one.

Reply:
We have many accounts and many roles per accounts, I realize the response was stable so I used the role index on the response as a shortcut. But, I agree that a more robust version should be to pass the accountId (Although it would be great if the OneLogin API could provide the account alias) and the role name.

Reply:
```Although it would be great if the OneLogin API could provide the account alias```

That isn't something that OneLogin API will support, It can't contain endpoints for each specific 3rd party vendor, there are tons.